maas/plugins/vg_check.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# Copyright 2014, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import division
+
+import argparse
+import shlex
+import subprocess
+
+
+from maas_common import metric
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+
+def run_command(arg):
+    proc = subprocess.Popen(shlex.split(arg),
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=False)
+
+    out, err = proc.communicate()
+    ret = proc.returncode
+    return ret, out, err
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser(
+        description='Check space in a volume group')
+    parser.add_argument('vgname', default='cinder-volumes', nargs='?',
```

Comment:
I wonder if it would be better to remove the default and fail if one isn't supplied. The plugin is being used with automation so the vg should always be passed to the script. If it doesn't get one something's gone wrong but the script will mask that because it will return metrics.